libcontainer/console.go

```diff
@@ -11,5 +17,59 @@ type Console interface {
 	Path() string
 
 	// Fd returns the fd for the master of the pty.
-	Fd() uintptr
+	File() *os.File
+}
+
+const (
+	TerminalInfoVersion = 1
+	TerminalInfoType    = "terminal"
+)
+
+// TerminalInfo is the structure which is passed as the non-ancillary data
+// in the sendmsg(2) call when runc is run with --console-socket. It
+// contains some information about the container which the console master fd
+// relates to (to allow for consumers to use a single unix socket to handle
+// multiple containers). This structure will probably move to runtime-spec
+// at some point. But for now it lies in libcontainer.
+type TerminalInfo struct {
+	// Version of the API.
+	Version int `json:"version"`
+
+	// Type of message (future proofing).
+	Type string `json:"type"`
```

Comment:
I mainly did this for debugging with `strace` purposes. I'm happy to change it, but `syncT` has shown us that explicit type values makes it easier to debug stuff. Is there a strong reason to change it